src/SFML/Network/IpAddress.cpp

```diff
@@ -37,68 +37,69 @@
 namespace sf
 {
 ////////////////////////////////////////////////////////////
-const IpAddress IpAddress::None;
 const IpAddress IpAddress::Any(0, 0, 0, 0);
 const IpAddress IpAddress::LocalHost(127, 0, 0, 1);
 const IpAddress IpAddress::Broadcast(255, 255, 255, 255);
 
 
 ////////////////////////////////////////////////////////////
-IpAddress::IpAddress() = default;
+std::optional<IpAddress> IpAddress::create(std::string_view address)
+{
+    using namespace std::string_view_literals;
 
+    if (address.empty())
+    {
+        return std::nullopt;
+    }
 
-////////////////////////////////////////////////////////////
-IpAddress::IpAddress(const std::string& address)
-{
-    if (address == "255.255.255.255")
+    if (address == "255.255.255.255"sv)
     {
         // The broadcast address needs to be handled explicitly,
         // because it is also the value returned by inet_addr on error
-        m_address = INADDR_BROADCAST;
-        return;
+        return IpAddress(NoByteOrderConversionTag{}, static_cast<Uint32>(INADDR_BROADCAST));
     }
 
-    if (address == "0.0.0.0")
+    if (address == "0.0.0.0"sv)
     {
-        m_address = INADDR_ANY;
-        return;
+        return IpAddress(NoByteOrderConversionTag{}, static_cast<Uint32>(INADDR_ANY));
     }
 
     // Try to convert the address as a byte representation ("xxx.xxx.xxx.xxx")
-    sf::Uint32 ip = inet_addr(address.c_str());
-    if (ip != INADDR_NONE)
+    if (const Uint32 ip = inet_addr(address.data()); ip != INADDR_NONE)
     {
-        m_address = ip;
-        return;
+        return IpAddress(NoByteOrderConversionTag{}, ip);
     }
 
     // Not a valid address, try to convert it as a host name
-    addrinfo hints;
-    std::memset(&hints, 0, sizeof(hints));
+    addrinfo hints{}; // Zero-initialize
     hints.ai_family = AF_INET;
+
     addrinfo* result = nullptr;
-    if (getaddrinfo(address.c_str(), nullptr, &hints, &result) == 0 && result)
+    if (getaddrinfo(address.data(), nullptr, &hints, &result) == 0 && result != nullptr)
     {
         sockaddr_in sin;
         std::memcpy(&sin, result->ai_addr, sizeof(*result->ai_addr));
-        ip = sin.sin_addr.s_addr;
+
+        const Uint32 ip = sin.sin_addr.s_addr;
         freeaddrinfo(result);
-        m_address = ip;
-        return;
+
+        return IpAddress(NoByteOrderConversionTag{}, ip);
     }
```

Comment:
Is the overload with `NoByteOrderConversionTag` necessary? It is only used internally yet requires being part of the externally visible API. Since this functionality is required from within the class itself, couldn't one just create a value that is initialized to e.g. `sf::IpAddress::Any` and manually set m_value in it? Any compiler that is reasonable at optimizing should be able to combine the statements into a single operation.